### DISCUSSION TITLE
Constrained tree search not working.

### DISCUSSION THREAD

**Discussion by ghost** [iqtree/iqtree2#258] [Q&A] [answered]
I played with my data, addressing all the common issues raised previously by others, but I could not get the constraint to work. I decided to try it on the example data but ran into the exact same problem. Could someone help? Thanks! 
[example.constr0.log](https://github.com/user-attachments/files/16091870/example.constr0.log)

**Comment by thomaskf** [comment, marked as answer]
Hi @zqtpck ,
The message in the log file is "ERROR: Taxon Seal in constraint tree does not appear in the full tree".
So it seems that there is a taxon "Seal" inside the input constraint tree, which, however, does not appear in the input alignment.

**Reply by ghost** [reply]
Thank you! I fixed the problem by correcting inconsistencies in the taxon names. I'm not sure how those errors were introduced initially. Now, I am struggling to get IQ-Tree to respect the constraint. The resulting tree has polytomies! I am using the UCE tree as a constraint on a four-gene tree.

**Reply by roblanf** [reply]
@zqtpck if you have another question, please start a new discussion. This keeps the discussion forum well organised so others can search. Also please read the new guide to posting, so that when you post your new question we have a good shot at answering it!
